pkg/restapi/webhook/webhook.go

```diff
@@ -16,44 +16,39 @@ import (
 	"github.com/hyperledger/aries-framework-go/pkg/common/log"
 )
 
-const (
-	notificationInterval    = 2 * time.Second
-	notificationSendTimeout = 10 * time.Second
-)
+const notificationSendTimeout = 10 * time.Second
 
 var logger = log.New("aries-framework/agentd")
 
-// StartWebhookDispatcher will launch the webhook dispatcher in a new goroutine.
-// The dispatcher is responsible for sending notifications to subscribers.
-func StartWebhookDispatcher(subscriberURLs []string) {
-	if len(subscriberURLs) == 0 {
-		logger.Warnf("No subscriber URLs provided. Webhook dispatcher will not start.")
-		return
-	}
+// Notifier represents a webhook dispatcher.
+type Notifier interface {
+	Notify(topic string, message []byte) error
+}
 
-	go dispatch(subscriberURLs)
+// HTTPNotifier is a webhook dispatcher capable of notifying multiple subscribers via HTTP.
+type HTTPNotifier struct {
+	WebhookURLs []string
 }
 
-func dispatch(subscriberURLs []string) {
-	// TODO: Data should should be pushed whenever a record is created or its state property is updated  (see #472).
-	ticker := time.NewTicker(notificationInterval)
-	defer ticker.Stop()
-	for range ticker.C {
-		for _, subscriberURL := range subscriberURLs {
-			err := sendNotification(subscriberURL)
-			if err != nil {
-				logger.Errorf(err.Error())
-			}
+// Notify sends the given message to all of the WebhookURLs.
+// Topic is appended to the end of the webhook (subscriber) URL. E.g. localhost:8080/topic
+// If multiple errors are encountered, then the first one is returned.
+func (n HTTPNotifier) Notify(topic string, message []byte) error {
+	var returnErr error
+	for _, webhookURL := range n.WebhookURLs {
+		err := notify(fmt.Sprintf("%s%s%s", webhookURL, "/", topic), message)
+		if returnErr == nil {
+			returnErr = err
```

Comment:
What if there are errors from multiple URLs? you show only one??

Reply:
Done; per your suggestion I now combine multiple errors.